### PR TITLE
Us#635  As a clinician, I would like to see the expected patient duration for the trial

### PR DESCRIPTION
**Pull request by snehi23**
**Taiga User Story**
Us#635  As a clinician, I would like to see the expected patient duration for the trial
**Taiga Task(s)**
#656 add patient's start and end date on trial page
**What did you change?**
I changed the query to fetch patient's list registered to trial. 
I changed the view to show fetched "Start" and "End" Date of patient   
**How can we test?**
1. go to any trial view
2. view the enlisted patient's data
3. you will find two extra fields [Start , End] as part of tabular view

screen shot 2016-04-22 at 10 37 18 am

### CODE REVIEW

File: controller/handler/trial.js [hunk 1]
```diff
@@ -11,6 +11,7 @@ const processComplianceCount = require('../helper/process-compliance-count');
 const processRules = require('../helper/process-rules');
 const processPatientStatus = require('../helper/process-patient-status');
 const httpNotFound = 404;
+const sqlDateFormat = 'ddd MMM DD YYYY HH:mm:ss ZZ';
```
Comment (poojaRal) [inline controller/handler/trial.js]: can you explain why do you require this date format?
Later at line 125 and 126, you do have a format applied to the date to get it in the required format?

**Comment by iamaureen** [comment]
LGTM

**Comment by iamaureen** [comment]
LGTM

File: controller/handler/trial.js (hunk 1 above)
Comment (snehi23) [inline controller/handler/trial.js]: "sqlDateFormat" is used to map the date format from mySQL to be used by moment.js
Later I formatted the date in "MM-DD-YYYY"

It seems code is working without it too.

Comment (snehi23) [inline controller/handler/trial.js]: @poojaRal I removed unnecessary formatting.

Comment (poojaRal) [inline controller/handler/trial.js]: I do no think this sqlDateFormat variable is required.

File: controller/handler/trial.js [hunk 2]
```diff
@@ -121,6 +122,11 @@ function trialView (request, reply) {
                     patient.totalMissed = 0;
                 }
 
+                patient.dateStarted = moment(patient.dateStarted, sqlDateFormat)
```
Comment (poojaRal) [inline controller/handler/trial.js]: Instead of these lines it should be -

patient.dateStarted = moment(patient.dateStarted).format('YYYY-MM-DD')
patient.dateCompleted = moment(patient.dateCompleted).format('YYYY-MM-DD')

You can refer to the dashboard.js controller file as to how IRB start & end dates are displayed there for a trial.

File: controller/handler/trial.js (hunk 1 above)
Comment (poojaRal) [inline controller/handler/trial.js]: Please remove the sqlDateFormat variable. I do not think we require it.
Instead use the below conversions -

patient.dateStarted = moment(patient.dateStarted).format(''YYYY-MM-DD')
patient.dateCompleted = moment(patient.dateCompleted).format('YYYY-MM-DD')

This is the same way for IRB start & end date population in dashboard controller. 
Please refer if required.

File: controller/handler/trial.js [hunk 3]
```diff
@@ -121,6 +121,11 @@ function trialView (request, reply) {
                     patient.totalMissed = 0;
                 }
 
+                patient.dateStarted = moment(patient.dateStarted)
+                    .format('MM-DD-YYYY');
```
Comment (poojaRal) [inline controller/handler/trial.js]: I think the format followed in the portal for dates is 'YYYY-MM-DD'

Comment (snehi23) [inline controller/handler/trial.js]: On patient's view, I had seen the format used as 'MM-DD-YYYY'.
I can change it if you want me to.

Comment (poojaRal) [inline controller/handler/trial.js]: Yes I see that on patient's view we are using format 'MM-DD'YYYY'
Then do not change it.
Let this be.
LGTM

**Comment by poojaRal** [comment]
LGTM

**Comment by ChristianMurphy** [comment]
LGTM
